packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,268 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { getFamilyTypes } from "../content";
+import { EntityType, IHubCatalog, IHubCollection } from "./types";
+
+export type WellKnownCatalog =
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+export type WellKnownCollection =
+  | "appAndMap"
+  | "content"
+  | "dataset"
+  | "document"
+  | "event"
+  | "feedback"
+  | "initiative"
+  | "project"
+  | "site"
+  | "solution"
+  | "template";
```

Comment:
I'm curious what other devs think, should we pluralize these? For example, I'm much more used to saying the `documents` collection rather than the `document` collection. And since we don't get auto complete in opendata-ui, I could see my mind adding the `s` in accidentally and wonder why it's not working.

Reply:
I don't have strong preferences... I checked that this does line up with the `HubFamily` type, which uses singular... I wonder if we can / should define `WellKnownCollection` in terms of `HubFamily`? 
e.g. `export type WellKnownCollection = Exclude<HubFamily, "app"> | "appAndMap";` which results in this
![image](https://user-images.githubusercontent.com/119129/217366766-b17c4511-27c2-4e42-a468-f51e05ac01e6.png) 

I think this is a good means to ensure consistency - but be sure to double check the entries in the two types... I think you will also need to add `solution` to `WellCknownCollection`

Reply:
Another option came to mind re: @sonofflynn89's comment about not knowing why something failed b/c of the lack of typing in Ember... for Capabilities I defined the valid values in an array and created the type from that...

![image](https://user-images.githubusercontent.com/119129/217371030-8acd3cf5-4430-447e-81e2-613a77653b23.png)

This allows for the creation of `isCapability` fn, which I use in the fn's which accept a `Capability` as an argument.